velox/substrait/SubstraitToVeloxPlanValidator.cpp

```diff
@@ -0,0 +1,336 @@
+  }
+  auto rowType = std::make_shared<RowType>(std::move(names), std::move(types));
+
+  // Validate the project expressions.
+  const auto& projectExprs = projectRel.expressions();
+  std::vector<std::shared_ptr<const core::ITypedExpr>> expressions;
+  expressions.reserve(projectExprs.size());
+  try {
+    for (const auto& expr : projectExprs) {
+      expressions.emplace_back(exprConverter_->toVeloxExpr(expr, rowType));
+    }
+    // Try to compile the expressions. If there is any unregistred funciton or
+    // mismatched type, exception will be thrown.
+    exec::ExprSet exprSet(std::move(expressions), execCtx_);
+  } catch (const VeloxException& err) {
+    VLOG(0) << "Validation failed for expression in ProjectRel due to:"
+            << err.message();
+    return false;
+  }
+  return true;
+}
+
+bool SubstraitToVeloxPlanValidator::validate(
+    const ::substrait::FilterRel& filterRel) {
+  // Not verified the Filter and fallback it currently.
+  return false;
+}
+
+bool SubstraitToVeloxPlanValidator::validate(
+    const ::substrait::AggregateRel& aggRel) {
+  if (aggRel.has_input() && !validate(aggRel.input())) {
+    return false;
+  }
+
+  // Validate input types.
+  if (aggRel.has_advanced_extension()) {
+    const auto& extension = aggRel.advanced_extension();
+    std::vector<TypePtr> types;
+    if (!validateInputTypes(extension, types)) {
+      VLOG(0) << "Validation failed for input types in AggregateRel";
+      return false;
+    }
+  }
+
+  // Validate groupings.
+  for (const auto& grouping : aggRel.groupings()) {
+    for (const auto& groupingExpr : grouping.grouping_expressions()) {
+      const auto& typeCase = groupingExpr.rex_type_case();
+      switch (typeCase) {
+        case ::substrait::Expression::RexTypeCase::kSelection:
+          break;
+        default:
+          VLOG(0) << "Only field is supported in groupings.";
+          return false;
+      }
+    }
+  }
+
+  // Validate aggregate functions.
+  std::vector<std::string> funcSpecs;
+  funcSpecs.reserve(aggRel.measures().size());
+  for (const auto& smea : aggRel.measures()) {
+    try {
+      const auto& aggFunction = smea.measure();
+      funcSpecs.emplace_back(
+          planConverter_->findFuncSpec(aggFunction.function_reference()));
+      toVeloxType(subParser_->parseType(aggFunction.output_type())->type);
+      for (const auto& arg : aggFunction.args()) {
+        auto typeCase = arg.rex_type_case();
+        switch (typeCase) {
+          case ::substrait::Expression::RexTypeCase::kSelection:
+            break;
+          default:
+            VLOG(0) << "Only field is supported in aggregate functions.";
+            return false;
+        }
+      }
+    } catch (const VeloxException& err) {
+      VLOG(0) << "Validation failed for aggregate function due to: "
+              << err.message();
+      return false;
+    }
+  }
+
+  std::unordered_set<std::string> supportedFuncs = {"sum", "count", "avg"};
```

Comment:
Why are these hard-coded here and not fetched from the registry of the aggregate functions, i.e. getAggregateFunctionSignatures or aggregateFunctions from velox/exec/Aggregate.h?